### PR TITLE
Feature/keep ratio while cropping

### LINKED ISSUE #42: [Bug] Crop Image by ratio not resizing image to width and height
After perform image crop by ratio, the width and height should correspond to the give crop ration, but for some reason the plugin still treat the final image width and height as square.

Needs fixing.

### PR DESCRIPTION
**Pull request by ikbendewilliam**
Fixed #42

### CODE REVIEW

File: lib/src/calculators/calculate_crop_fit_params.dart [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  final double cropSizeWidth;
+
+  /// the height of the area to crop
+  final double cropSizeHeight;
+
+  /// used to adjust image scale
+  final double defaultScale;
+
+  switch (imageFit) {
+    case CustomImageFit.fillCropSpace:
```
Comment (vanlooverenkoen) [inline lib/src/calculators/calculate_crop_fit_params.dart:26]: I think I would also move this to separate classes

File: lib/src/calculators/calculate_on_crop_params.dart [hunk 2]
```diff
@@ -9,108 +9,135 @@ OnCropParams caclulateOnCropParams({
...
+  /// Temp variable used to calculate the translateScale
+  final double uiSize;
 
   switch (imageFit) {
-    case CustomImageFit.fitCropSpace:
-      final uiSize = min(screenWidth, screenHeight);
-      final cropSize = max(imageWidth, imageHeight).toDouble();
-      final translateScale = cropSize / (uiSize * cropPercentage);
-      final scale = dataScale;
+    case CustomImageFit.fillCropSpace:
```
Comment (vanlooverenkoen) [inline lib/src/calculators/calculate_on_crop_params.dart:30]: same comment as above

File: lib/src/calculators/calculate_crop_fit_params.dart (hunk 1 above)
Comment (ikbendewilliam) [reply lib/src/calculators/calculate_crop_fit_params.dart:26]: I disagree, it's much easier to see the other implementations directly 🤔

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,50 +1,55 @@
+## [0.0.12] - 2023-09-08
```
[the remark this answers is not in the document]
Comment (ikbendewilliam) [reply CHANGELOG.md:1]: There not actually https://pub.dev/packages/custom_image_crop/changelog#006

**Review by jorre127 — APPROVED** [review]

**Pull request merged**